apollo-runtime/src/commonMain/kotlin/com/apollographql/apollo3/network/http/HttpNetworkTransport.kt

```diff
@@ -67,32 +73,41 @@ private constructor(
         )
       }
 
-      // do not capture request
-      val operation = request.operation
-      val response = worker.doWork {
-        try {
-          operation.parseJsonResponse(
-              jsonReader = httpResponse.body!!.jsonReader(),
-              customScalarAdapters = customScalarAdapters
-          )
-        } catch (e: Exception) {
-          throw wrapThrowableIfNeeded(e)
-        }
+      // When using @defer, the response contains multiple parts, using the multipart content type.
+      // See https://github.com/graphql/graphql-over-http/blob/main/rfcs/IncrementalDelivery.md
+      val bodies: Flow<BufferedSource> = if (httpResponse.isMultipart) {
+        multipartBodyFlow(httpResponse)
+      } else {
+        flowOf(httpResponse.body!!)
       }
 
-      @Suppress("DEPRECATION")
-      emit(
-          response.newBuilder()
-              .requestUuid(request.requestUuid)
-              .addExecutionContext(
-                  HttpInfo(
-                      startMillis = millisStart,
-                      endMillis = currentTimeMillis(),
-                      statusCode = httpResponse.statusCode,
-                      headers = httpResponse.headers
-                  )
-              )
-              .build()
+      emitAll(
+          bodies.map { body ->
+            val operation = request.operation
```

Comment:
```suggestion
            // do not capture request
            val operation = request.operation
```

I'm not sure why this was important but I'd keep the comment for context